crates/consensus/src/header.rs

```diff
@@ -346,6 +375,50 @@ impl Header {
     pub const fn seal(self, hash: B256) -> Sealed<Self> {
         Sealed::new_unchecked(self, hash)
     }
+
+    /// Validates that the EIP-4844 fields in the current block header are correct compared to the
+    /// parent block header.
+    ///
+    /// Specifically, this function checks two things:
+    /// 1. The current block header contains the `blob_gas_used` and `excess_blob_gas` fields.
+    /// 2. The `excess_blob_gas` field matches the expected value calculated from the parent header
+    ///    fields.
+    pub fn validate_against_parent_4844(&self, parent: &Self) -> Result<(), ConsensusError> {
```

Comment:
Wanted to create a follow up PR with unit tests to validate all the scenarios, but first I wanted to know if this proposal of migration would be accepted